pkg/stream/xprotocol/connpool_binding.go

```diff
@@ -81,7 +81,7 @@ func (p *poolBinding) NewStream(ctx context.Context, receiver types.StreamReceiv
 
 	downstreamConn := getDownstreamConn(ctx)
 	c.downstreamConn = downstreamConn
-	downstreamConn.AddConnectionEventListener(downstreamCloseListener{upstreamClient: c})
+	c.addDownConnListenerOnce()
```

Comment:
change addDownConnListenerOnce to addDownConnOnce(ctx), do the downstreamConn bind once too?

Reply:
```
c.addDownConnOnce(ctx)

func addDownConnOnce(ctx context.Context) {
      once.Do(func(){
       downstreamConn := getDownstreamConn(ctx)
	c.downstreamConn = downstreamConn
ac.downstreamConn.AddConnectionEventListener(downstreamCloseListener{upstreamClient: ac})
 })
}
```